src/SvmUtils.ts

```diff
@@ -55,28 +55,33 @@ export async function readEvents<IDL extends Idl = Idl>(
     maxSupportedTransactionVersion: 0,
   });
 
-  let eventAuthorities = new Map();
+  if (txResult === null) return [];
+
+  const eventAuthorities: Map<string, PublicKey> = new Map();
   for (const program of programs) {
     eventAuthorities.set(
       program.programId.toString(),
-      findProgramAddress("__event_authority", program.programId).publicKey.toString()
+      findProgramAddress("__event_authority", program.programId).publicKey
     );
   }
 
-  let events: any[] = [];
+  const events = [];
 
-  // TODO: Add support for version 0 transactions.
-  if (!txResult || txResult.transaction.message.version !== "legacy") return events;
+  // Resolve any potential addresses that were passed from address lookup tables.
+  const messageAccountKeys = txResult.transaction.message.getAccountKeys({
```

Comment:
This is still capturing the accounts not passed in ALT right?

Reply:
yes, it covers both static accounts and also the ones passed in ALT (see accountKeysFromLookups prop below)